command/root.go

```diff
@@ -250,6 +250,30 @@ func determineBaseRepo(cmd *cobra.Command, ctx context.Context) (ghrepo.Interfac
 	return baseRepo, nil
 }
 
+func convertRepoInterfaceToRepository(ctx context.Context, repo ghrepo.Interface) (*api.Repository, error) {
+	apiClient, err := apiClientForContext(ctx)
```

Comment:
We should try to pass an API client from the outside to avoid instantiating multiple API clients during the lifetime of the same command https://github.com/cli/cli/pull/943